### PR TITLE
[ADD] Use data_link helper for qc_panel

### PR DESCRIPTION
**Pull request by natacha-beck**
Use the Subviewer mechanism in order to display files in QC_panel if a viewer is available.

### CODE REVIEW

File: BrainPortal/app/helpers/userfiles_helper.rb [hunk 1]
```diff
@@ -100,10 +103,10 @@ def data_link(file_name, userfile)
                                              :viewer                => viewer,
                                              :viewer_userfile_class => matched_class
                                              ),
-            :replace => "sub_viewer_filecollection_cbrain",
+            :replace => replace_div_id,
           }
         ) do
-          ("<span class=\"sub_viewable_link\">"+display_name+"</span>").html_safe
+          ("<span class=\"sub_viewable_link "+replace_div_id+"\">"+display_name+"</span>").html_safe
```
Comment (prioux) [inline BrainPortal/app/helpers/userfiles_helper.rb:109]: ID vs CLASS to resolve.

Comment (natacha-beck) [reply BrainPortal/app/helpers/userfiles_helper.rb]: Should be ok now.

File: BrainPortal/app/views/userfiles/_default_qc_panel.html.erb [hunk 2]
```diff
@@ -22,15 +22,29 @@
...
+      </span>
+    <% end %>
   <% end %>
-<% end %>
+</div>
+
+<script type="application/javascript">
+  // Find the link inside the div and click it
+  var subViewerDiv  = $("#<%= div_id_to_replace %>");
+  var subViewerLink = subViewerDiv.find(".ajax_onclick_show_element");
```
Comment (prioux) [inline BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:46]: What is this class? How is it added to the HTML elements? If it's done later after the page is loaded, could this script be executed too early? Maybe the script tag shoudl use the attribute `defer` ?

Comment (prioux) [reply BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:46]: Try this in a bran new browser in incognito and no cached javascript

Comment (natacha-beck) [reply BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:46]: Done, it works fine in an incognito browser.

File: BrainPortal/app/views/userfiles/_default_qc_panel.html.erb [hunk 3]
```diff
@@ -22,15 +22,29 @@
...
+  <% if ! @userfile.is_locally_synced? %>
+    This file must be synced locally to view QC data.
   <% else %>
-    <span class="warning">
-    This file does not seem to contain any QC data, or no template is available to QC this file type.
-    </span>
+    <% link = data_link(@userfile.name, @userfile, div_id_to_replace) %>
+    <% if link =~ /data-url=/%>
+        <%= link %>
+      </div>
```
Comment (prioux) [inline BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:34]: Your DIV structure is imbalanced!!!

File: BrainPortal/app/views/userfiles/_quality_control_panel.html.erb [hunk 4]
```diff
@@ -36,10 +36,12 @@
     <legend>Navigation: file <%= @current_index + 1 %> out of <%= @filelist.size %></legend>
     <%= link_to_userfile_if_accessible(@userfile, current_user, :html_options => { :target => '_blank' }) %>
     <p>
-    <% if @current_index > 0 %>
+    <% first_file = @current_index > 0 %>
```
Comment (prioux) [inline BrainPortal/app/views/userfiles/_quality_control_panel.html.erb:39]: I don't get it. The variable 'first_file' is TRUE when this is NOT the first file ?!?!? What kind of naming is this?

**Review by prioux — CHANGES_REQUESTED** [review]
Fix bugs please

**Review by prioux — CHANGES_REQUESTED** [review]
See comments

File: BrainPortal/app/views/userfiles/_default_qc_panel.html.erb [hunk 5]
```diff
@@ -23,14 +23,30 @@
...
     <span class="warning">
-    This file does not seem to contain any QC data, or no template is available to QC this file type.
+      This file does not seem to contain any QC data, or no template is available to QC this file type.
     </span>
   <% end %>
 <% end %>
 
+<script type="application/javascript" defer>
+  // Find the link inside the div and click it
+  <% if div_id_to_replace %>
```
Comment (prioux) [inline BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:45]: I feel uncomfortable about the scope of the variable `id_to_replace` though. It feels it is a local variable inside a code block well above in the HTML page, so maybe move its initialization out to be at the level of the entire page, or (better) move the entire script tag inside the `if link =~ /data-url/`

File: BrainPortal/app/views/userfiles/_default_qc_panel.html.erb [hunk 6]
```diff
@@ -23,14 +23,30 @@
...
-  This file must be synced locally to view QC data.
+  <span class="warning">
+    This file must be synced locally to view QC data.
+  </span>
 <% else %>
-  <% if @userfile.is_a?(ImageFile) %>
-    <%= image_tag url_for(:action  => :content, :id => @userfile.id) %><br>
+  <% div_id_to_replace = "sub_viewer_#{@userfile.id}" %>
+  <% link = data_link(@userfile.name, @userfile, div_id_to_replace) %>
+  <% if link =~ /data-url=/%>
```
Comment (prioux) [inline BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:32]: Please add a space before the final `%>`

Comment (prioux) [reply BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:32]: Also, here would go the `<script>` tag

File: BrainPortal/app/views/userfiles/_default_qc_panel.html.erb (hunk 5 above)
Comment (prioux) [reply BrainPortal/app/views/userfiles/_default_qc_panel.html.erb:45]: My original comment before I suggested moving the script tag was that the condition `if div_id_to_replace` shoudl go AROUND the entire script tag, but that is moot if you move the script tag to the HTML above.

**Review by prioux — CHANGES_REQUESTED** [review]
Re-entered some comments

**Pull request merged**